src/views/virtualmachines/list/metrics.ts

```diff
@@ -41,3 +44,39 @@ export const setVMCPURequested = (vmName: string, vmNamespace: string, cpuReques
   const vmMetrics = getVMMetrics(vmName, vmNamespace);
   vmMetrics.cpuRequested = cpuRequested;
 };
+
+export const getCPUUsagePercentage = (vmName: string, vmNamespace: string) => {
+  const { cpuRequested, cpuUsage } = getVMMetrics(vmName, vmNamespace);
+
+  if (isEmpty(cpuRequested) || isEmpty(cpuUsage)) return;
+
+  const percentage = (cpuUsage * 100) / cpuRequested;
+  return percentage;
+};
+
+export const getMemoryUsagePercentage = (
+  vmName: string,
+  vmNamespace: string,
+  vmiMemory: string,
+) => {
+  const { memoryUsage } = getVMMetrics(vmName, vmNamespace);
+
+  if (isEmpty(memoryUsage) || isEmpty(vmiMemory)) return;
+
+  const memoryRequested = getMemorySize(vmiMemory);
+
+  const memoryAvailableBytes = xbytes.parseSize(
+    `${memoryRequested?.size} ${memoryRequested?.unit}B`,
+  );
+
+  const percentage = (memoryUsage * 100) / memoryAvailableBytes;
+  return percentage;
+};
+
+export const getNetworkUsagePercentage = (vmName: string, vmNamespace: string) => {
+  const { networkUsage } = getVMMetrics(vmName, vmNamespace);
+
+  if (isEmpty(networkUsage)) return;
```

Comment:
do u need this check here?

Reply:
Yes
`1 - undefined` and `undefined - 1` have the same result and it makes the sort function asymmetrical

Reply:
not really

Reply:
removed